contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,302 @@
+
+Identified use-cases aim to ensure that administrators have a way to restrict users or namespaces, it allows administrators to:
+
+-  Restrict execution for specific applications (which are namespace scoped) into certain nodes
+-  Create policies that prevent users from even attempting to schedule workloads onto masters to maximize security
+-  require that a pods under a namespace run on dedicated nodes
+-  Restrict usage of some `PriorityClass`
+-  Restrict usage to a specific set of schedulers.
+-  enforcing pod affinity or anti-affinity rules on some particular namespace.
+
+Also Mandatory values must also be enforced by scheduling policies in case of:
+
+-  mutli-arch (amd64, arm64) of multi-os (Linux, Windows) clusters (will also be handled later by the [RuntimeClass]() KEP)
+-  multi-az / region / failure domain clusters
+
+# Overview
+
+### syntaxic standards
+
+### empty match and unset fields:
+
+We follow Kubernetes conventions, an empty field matches everything an unset one matches nothing
+
+### how policies are computed
+
+Policies are computed using the following algorithm:
+
+```
+sortedPolicies = sort_by_priority(sort_deny_first(policies))
+for policy in sortedPolicies:
+  if policy matches pod: // all specified policy rules match
+    return policy.action
+```
+ note that rules of policies from a lower priority are superseeded by ones from a higher priority if they match.
+
+
+
+# Detailed Design
+
+
+## SchedulingPolicy
+
+Proposed API group: `policy/v1alpha1`
+
+
+### SchedulingPolicy content
+
+SchedulingPolicy spec is composed of optional fields that allow scheduling rules. If a field is absent from a SchedulingPolicy, this `SchedulingPolicy` won't allow any item from the missing field.
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  priority: <exception,cluster,default>
+  action:  <allow,deny>     
+  rules:
+    namespaceSelector:
+      - key: value
+        operator: operator
+    podSelector:
+        key: value
+    schedulerNames:
+       match: []
+    priorityClassNames:
+       match: []
+    tolerations:
+       match: []
+    nodeSelectors:
+       match: []
+    nodeAffinities:
+       match: []
+    podAntiAffinities:
+       match: []
+    podAffinities:
+       match: []
+```
+
+- allow: When a policy matches the pod is allowed
+- deny: when a policy matches the pod is denied
+
+
+### Scoping
+
+The `spec.rules.namespaceSelector` and `spec.rules.podSelector` attributes scope the policy application range.
+
+Pods must match both `spec.namespaceSelector` and `spec.podSelector` matching rules to be constrained by a policy.
+Both `spec.namespaceSelector` and `spec.podSelector` are optional. If absent, all pods and all namespaces are targeted by the policy.
+
+### Scheduler name
+
+If `schedulerNames` is absent from `allowed`, `default` or `required`, no scheduler is allowed by this specific policy.
```

Comment:
required is no more?

I'm not sure what this means... see my comment above about matching the empty value.

> no scheduler is allowed by this specific policy.

If this is a common use case, we may want a way to express "any value that is set" (i.e. "deny any value that is set")

Reply:
Hmm, I guess that would be a use for `noneOf` which I said I didn't think we need...

Reply:
+1 for use cases and the actual policy/policies needed to accomplish the use case. bonus points for reasoning through how those policies would interact with other policies present